airflow/stats.py

```diff
@@ -191,93 +443,121 @@ def test(self, stat):
             return True  # default is all metrics allowed
 
 
+def _format_safe_stats_logger_args(func):
+    @functools.wraps(func)
+    def func_wrap(_self, stat, *args, labels=None, **kwargs):
+        if labels:
+            # Remove {} from stat key e.g. {class_name} => class_name
+            stat = stat.format(**labels)
+        return validate_stat(func)(_self, stat, *args, **kwargs)
+
+    return func_wrap
+
+
 class SafeStatsdLogger:
     """Statsd Logger"""
 
     def __init__(self, statsd_client, allow_list_validator=AllowListValidator()):
         self.statsd = statsd_client
         self.allow_list_validator = allow_list_validator
 
-    @validate_stat
-    def incr(self, stat, count=1, rate=1):
+    @_format_safe_stats_logger_args
```

Comment:
I'd rather we didn't have to do string processing every time we emitted a metric, given this is "static"/precomputable.

How much work would it be to have this done once in get_statsd_logger/get_dogstatsd_logger?

Reply:
Oh, it is for dogstatsd, but not for statsd.

Reply:
Edit: I thought that we were using the METRIC_LIST to emit stats (hence the pre-computing would be possible) but we aren't, so this can't be done.